modules/@apostrophecms/form-widget/ui/src/index.js

```diff
@@ -169,31 +172,12 @@ export default () => {
         input.queryParams = apos.http.parseQuery(window.location.search);
       }
 
-      async function uploadFiles(field, fieldInput) {
-        const fileIds = [];
-        const uploadRoute = '/api/v1/@apostrophecms/form-widget/upload';
-
+      async function appendFiles(field, fieldInput, data) {
+        let i = 0;
         for (const file of fieldInput.files) {
-          const formData = new window.FormData();
-          formData.append('field', field);
-          formData.append('file', file);
-
-          try {
-            const attachment = await apos.http.post(uploadRoute, {
-              busy: true,
-              body: formData
-            });
-
-            fileIds.push(attachment._id);
-          } catch (error) {
-            // Arranging for field level error display.
-            error.data = error.body.data;
-
-            throw error;
-          }
+          data.append(`${field}-${i}`, file);
```

Comment:
Fields are flat, never nested, so this is enough I assume?

Reply:
correct